core/amber/src/main/python/core/architecture/managers/statistics_manager.py

```diff
@@ -21,11 +22,26 @@ def get_statistics(self) -> typing.Tuple[int, int]:
             - self._control_processing_time,
         )
 
-    def increase_input_tuple_count(self) -> None:
-        self._input_tuple_count += 1
-
-    def increase_output_tuple_count(self) -> None:
-        self._output_tuple_count += 1
+    def increase_input_tuple_count(self, port_id: PortIdentity) -> None:
+        if port_id is None:
+            port = 0
```

Comment:
from the function signature, `port_id: PortIdentity` indicates it cannot be `None` value. If you do want to allow `None`, then please use `port_id: Optional[PortIdentity]`. 


For this case, I suggest we do not allow None and requires the caller to pass a valid `PortIdentity`.